lib/hooks/blueprints/actions/add.js

```diff
@@ -23,7 +23,15 @@ module.exports = function addToCollection (req, res) {
   // Set the blueprint action for parseBlueprintOptions.
   req.options.blueprintAction = 'add';
 
-  var queryOptions = parseBlueprintOptions(req);
+  var queryOptions;
+  try {
+    queryOptions = parseBlueprintOptions(req);
+  } catch (err) {
+    switch (err.name) {
```

Comment:
why switch? can this be changed to
catch (err) {
if (err.name === 'UsageError') {
  return res.badRequest(formatUsageError(err, req));
}

return res.serverError(err);
}

Or
return err.name === 'UsageError' ? res.badRequest(formatUsageError(err, req)) : res.serverError(err)

This is much cleaner. I think you don't have multiple if...else so switch is not right option here.